go/externals/proof_support_facebook.go

```diff
@@ -0,0 +1,229 @@
+
+	if !strings.HasPrefix(strings.ToLower(h.GetAPIURL()), wantedURL) {
+		return libkb.NewProofError(keybase1.ProofStatus_BAD_API_URL,
+			"Bad hint from server; URL should start with '%s'", wantedURL)
+	}
+
+	// TODO: We could ignore this portion of the server's hint. Should we?
+	if !strings.Contains(h.GetCheckText(), wantedMediumID) {
+		return libkb.NewProofError(keybase1.ProofStatus_BAD_SIGNATURE,
+			"Bad proof-check text from server; need '%s' as a substring", wantedMediumID)
+	}
+
+	return nil
+}
+
+func (rc *FacebookChecker) ScreenNameCompare(s1, s2 string) bool {
+	s1NoDots := strings.Replace(s1, ".", "", -1)
+	s2NoDots := strings.Replace(s2, ".", "", -1)
+	return libkb.Cicmp(s1NoDots, s2NoDots)
+}
+
+func (rc *FacebookChecker) CheckStatus(ctx libkb.ProofContext, h libkb.SigHint) libkb.ProofError {
+	if pvl.UsePvl {
+		return pvl.CheckProof(ctx, pvl.GetHardcodedPvl(), keybase1.ProofType_FACEBOOK, rc.proof, h)
+	}
+	return rc.CheckStatusOld(ctx, h)
+}
+
+func (rc *FacebookChecker) CheckStatusOld(ctx libkb.ProofContext, h libkb.SigHint) libkb.ProofError {
+	apiURL := h.GetAPIURL()
+	res, err := ctx.GetExternalAPI().GetHTML(libkb.NewAPIArg(apiURL))
+	if err != nil {
+		return libkb.XapiError(err, apiURL)
+	}
+
+	username, proofErr := extractUsername(res.GoQuery)
+	if proofErr != nil {
+		return proofErr
+	}
+
+	if !usernamesEqual(username, rc.proof.GetRemoteUsername()) {
+		return libkb.NewProofError(keybase1.ProofStatus_BAD_USERNAME, "Usernames don't match: '%s' != '%s'", username, rc.proof.GetRemoteUsername())
+	}
+
+	proofText, proofErr := extractProofText(res.GoQuery)
+	if proofErr != nil {
+		return proofErr
+	}
+
+	if strings.TrimSpace(proofText) != strings.TrimSpace(h.GetCheckText()) {
+		return libkb.NewProofError(keybase1.ProofStatus_TEXT_NOT_FOUND, "Proof text not found: '%s' != '%s'", proofText, h.GetCheckText())
+	}
+
+	// The proof is good.
+	return nil
+}
+
+func extractUsername(doc *goquery.Document) (string, libkb.ProofError) {
+	selector := "#m_story_permalink_view  a"
+	usernameAnchor := doc.Find(selector)
+	if usernameAnchor.Length() == 0 {
+		return "", libkb.NewProofError(keybase1.ProofStatus_FAILED_PARSE, "Couldn't find username anchor tag $(%s)", selector)
+	}
+	// Only consider the first
+	usernameAnchor = usernameAnchor.First()
+
+	usernameLink, ok := usernameAnchor.Attr("href")
+	if !ok {
+		return "", libkb.NewProofError(keybase1.ProofStatus_FAILED_PARSE, "Couldn't find username href.")
+	}
+
+	parsedLink, err := url.Parse(usernameLink)
+	if err != nil {
+		return "", libkb.NewProofError(keybase1.ProofStatus_FAILED_PARSE, "Failed to parse username URL: %s", err)
+	}
+	splitPath := strings.Split(parsedLink.Path, "/")
+	if len(splitPath) < 1 {
+		return "", libkb.NewProofError(keybase1.ProofStatus_FAILED_PARSE, "Username URL has no path.")
+	}
+	username := splitPath[1]
+	return username, nil
+}
+
+func extractProofText(doc *goquery.Document) (string, libkb.ProofError) {
+	selector := "#m_story_permalink_view h3"
```

Comment:
You're right, I think the current logic can be exploited to make a post that's *not* a proof appear to be a proof, by commenting on it. Let's look together.